library/src/main/java/com/pokegoapi/auth/PtcCredentialProvider.java

```diff
@@ -165,32 +170,30 @@ private void login(String username, String password, int attempt)
 			Moshi moshi = new Moshi.Builder().build();
 
 			PtcAuthJson ptcAuth;
+			String response;
 			try {
-				String response = getResponse.body().string();
+				response = getResponse.body().string();
 				ptcAuth = moshi.adapter(PtcAuthJson.class).fromJson(response);
 			} catch (IOException e) {
 				throw new LoginFailedException("Looks like the servers are down", e);
 			}
 
-			HttpUrl url = HttpUrl.parse(LOGIN_URL).newBuilder()
-					.addQueryParameter("lt", ptcAuth.getLt())
-					.addQueryParameter("execution", ptcAuth.getExecution())
-					.addQueryParameter("_eventId", "submit")
-					.addQueryParameter("username", username)
-					.addQueryParameter("password", password)
+			FormBody url = new Builder()
+					.add("lt", ptcAuth.getLt())
+					.add("execution", ptcAuth.getExecution())
+					.add("_eventId", "submit")
+					.add("username", username)
+					.add("password", password)
 					.build();
-
-			RequestBody reqBody = RequestBody.create(null, new byte[0]);
-
 			Request postRequest = new Request.Builder()
-					.url(url)
-					.method("POST", reqBody)
+					.url(LOGIN_URL)
+					.method("POST", url)
 					.build();
 
 			// Need a new client for this to not follow redirects
-			Response response;
+			Response response1;
```

Comment:
Could be better to refactor each login step into a new method so we don't need variable names like this. Otherwise, you could give it a more descriptive name.